### ISSUE TITLE
[PRE REVIEW]: TEAM: Tensegrity Engineering Analysis Master

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1507]
**Submitting author:** @ramaniitrgoyal92 (<a href="http://orcid.org/0000-0002-8128-3051">Raman Goyal</a>)
**Repository:** <a href="https://github.com/ramaniitrgoyal92/Tensegrity_Engineering_Analysis_Master-TEAM-" target ="_blank">https://github.com/ramaniitrgoyal92/Tensegrity_Engineering_Analysis_Master-TEAM-</a>
**Version:** v2.1
**Editor:** @Kevin-Mattheus-Moerman
**Reviewers:** @apsabelhaus, @vaishnavtv, @ctdegroot

**Author instructions**

Thanks for submitting your paper to JOSS @ramaniitrgoyal92. **Currently, there isn't an JOSS editor assigned** to your paper.

@ramaniitrgoyal92 if you have any suggestions for potential reviewers then please mention them here in this thread. In addition, [this list of people](https://bit.ly/joss-reviewers) have already agreed to review for JOSS and may be suitable for this submission.

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks.

For a list of things I can do to help you, just type:

```
@whedon commands
```

**What happens now?**

This submission is currently in a `pre-review` state which means we are waiting for an editor to be assigned and for them to find some reviewers for your submission. This may take anything between a few hours to a couple of weeks. Thanks for your patience :smile_cat:

You can help the editor by looking at [this list of potential reviewers](https://bit.ly/joss-reviewers) to identify individuals who might be able to review your submission (please start at the bottom of the list). Also, feel free to suggest individuals who are not on this list by mentioning their GitHub handles here.

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1507 with the following error: 

 Can't find any papers to compile :-(

**Comment by arfon** [comment]
Hi @ramaniitrgoyal92, thanks for your submission to JOSS. Please rename your paper to `paper.md`

**Comment by ramaniitrgoyal92** [comment]
> **Submitting author:** @ramaniitrgoyal92 ([Raman Goyal](http://orcid.org/0000-0002-8128-3051))
> **Repository:** https://github.com/ramaniitrgoyal92/Tensegrity_Engineering_Analysis_Master-TEAM-
> **Version:** v2.1
> **Editor:** Pending
> **Reviewer:** Pending
> 
> **Author instructions**
> 
> Thanks for submitting your paper to JOSS @ramaniitrgoyal92. **Currently, there isn't an JOSS editor assigned** to your paper.
> 
> @ramaniitrgoyal92 if you have any suggestions for potential reviewers then please mention them here in this thread. In addition, [this list of people](https://bit.ly/joss-reviewers) have already agreed to review for JOSS and may be suitable for this submission.
> 
> **Editor instructions**
> 
> The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:
> 
> ```
> @whedon commands
> `

The suggested reviewers:

@apsabelhaus
@moorepants
@ctdegroot
@vaishnavtv
@datseris

**Comment by apsabelhaus** [comment]
For the editor, once they're assigned:
If chosen as a reviewer, I'd be happy to review this work.

However, I'd like to quickly check about JOSS' Conflict of Interest policy (https://joss.readthedocs.io/en/latest/reviewer_guidelines.html), since I'm still new to open review (not blind). The submitting author @ramaniitrgoyal92 and I are professional acquaintances who have discussed research before and compared our past publications, but have not collaborated on funded work or authorship, and are not members of the same institution. Normally this wouldn't be a problem for single-blind review, but does it pose a challenge for open review? Either way, I believe I can provide an "impartial assessment," but would like to disclose our relationship before agreeing to be a reviewer.

**Comment by Datseris** [comment]
Hello. Thank you for invitation but I do not have the expertise to review this:

* I do not know anything about MATLAB (I actually have never _seen_ MATLAB code)
* I do not have the scientific expertise on the area the submitted `TEAM` software is tackling, which seems to be around control systems.

I will have to pass on this, sorry.

**Comment by arfon** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1507 with the following error: 

 /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-18b9a37e7f54/lib/whedon/orcid_validator.rb:12:in `initialize': undefined method `strip' for nil:NilClass (NoMethodError)
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-18b9a37e7f54/lib/whedon.rb:123:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-18b9a37e7f54/lib/whedon.rb:123:in `block in check_orcids'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-18b9a37e7f54/lib/whedon.rb:121:in `each'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-18b9a37e7f54/lib/whedon.rb:121:in `check_orcids'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-18b9a37e7f54/lib/whedon.rb:81:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-18b9a37e7f54/lib/whedon/processor.rb:36:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-18b9a37e7f54/lib/whedon/processor.rb:36:in `set_paper'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-18b9a37e7f54/bin/whedon:55:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-18b9a37e7f54/bin/whedon:116:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon assign @Kevin-Mattheus-Moerman as editor

**Comment by whedon** [comment]
OK, the editor is @Kevin-Mattheus-Moerman

**Comment by Kevin-Mattheus-Moerman** [comment]
@apsabelhaus thanks for raising the conflict of interest issue. I'll add you as a reviewer but will also recruit others. Thanks

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon assign @apsabelhaus as reviewer

**Comment by whedon** [comment]
OK, the reviewer is @apsabelhaus

**Comment by Kevin-Mattheus-Moerman** [comment]
@vaishnavtv @ctdegroot since you were involved with https://github.com/openjournals/joss-reviews/issues/1042, would you be interested in helping with this review?

**Comment by Kevin-Mattheus-Moerman** [comment]
@ramaniitrgoyal92 Please rename your paper to `paper.md`, this way `@whedon` will find it okay to render the PDF. Thanks

**Comment by ctdegroot** [comment]
@Kevin-Mattheus-Moerman Yes, I can help with this review.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedeon add @ctdegroot as reviewer

**Comment by Kevin-Mattheus-Moerman** [comment]
@vaishnavtv @ctdegroot we have to iron out some things with the paper and I hope to recruit an additional reviewer. I'll ping you again when we are ready to start the review process.

**Comment by vaishnavtv** [comment]
@Kevin-Mattheus-Moerman, I would love to help with the review, but under full disclosure, I'd like to state that the submitting author and I are personally acquainted and are pursuing a graduate degree from the same institution. We have not officially collaborated on any research, but our advisors have worked together in the past. If this violates any reviewer policies based on a conflict of interest, kindly let me know.

**Comment by Kevin-Mattheus-Moerman** [comment]
Thanks @vaishnavtv (sorry I accidentally tagged you in the comment above I just edited) for reporting that potential conflict of interest. If you'd like to help your help is still appreciated. I will however recruit an additional reviewer.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1507 with the following error: 

 /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-18b9a37e7f54/lib/whedon/orcid_validator.rb:12:in `initialize': undefined method `strip' for nil:NilClass (NoMethodError)
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-18b9a37e7f54/lib/whedon.rb:123:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-18b9a37e7f54/lib/whedon.rb:123:in `block in check_orcids'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-18b9a37e7f54/lib/whedon.rb:121:in `each'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-18b9a37e7f54/lib/whedon.rb:121:in `check_orcids'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-18b9a37e7f54/lib/whedon.rb:81:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-18b9a37e7f54/lib/whedon/processor.rb:36:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-18b9a37e7f54/lib/whedon/processor.rb:36:in `set_paper'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-18b9a37e7f54/bin/whedon:55:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-18b9a37e7f54/bin/whedon:116:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by ramaniitrgoyal92** [comment]
@whedon  generate pdf

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01507/joss.01507/10.21105.joss.01507.pdf)

**Comment by apsabelhaus** [comment]
great, happy to be a reviewer. @ramaniitrgoyal92 I'll probably need at least three weeks to get around to this - my dissertation draft is due to my committee ASAP - so you can expect my first round of comments during mid-July. @Kevin-Mattheus-Moerman let me know when the process is ready to start and that checklist is up. :+1:

**Comment by ramaniitrgoyal92** [comment]
@abelsiqueira Sure, take your time, really appreciate!

**Comment by Kevin-Mattheus-Moerman** [comment]
@Jfriesen222 would you be interested in reviewing this work on tensegrity modelling for [JOSS](http://joss.theoj.org)? The review process will focus on [the software](https://github.com/ramaniitrgoyal92/Tensegrity_Engineering_Analysis_Master-TEAM-) and [this short paper](https://github.com/openjournals/joss-papers/blob/joss.01507/joss.01507/10.21105.joss.01507.pdf).

**Comment by apsabelhaus** [comment]
@Kevin-Mattheus-Moerman is the review thread started for this submission yet? I'm available to start the review process.

**Comment by apsabelhaus** [comment]
@ramaniitrgoyal92 While we're waiting for an official review to be opened, I've got a few suggestions from a first look at the code. First, the paper is missing references [???], probably a bug in the latex.

I would recommend that the authors take a second look at some of the naming in the repository. As an example, the name:

Tensegrity Engineering Analysis Master

is a nice acronym, but is relatively vague as to its purpose. Specifically, "engineering analysis" is very broad. It looks like this library does statics and dynamics, but there's lots of other analysis for tensegrity structures that might be considered. A more specific name related to exactly the library's purpose would be helpful: example, how does this software differ from STEDY (https://github.com/isrlab/stedy)? This would help with the "purpose" requirement for JOSS.

Similarly, the words "main" and "master" don't have much meaning to me. "Master" as in master branch? "Main" as in "core components of the library?" The "Master" in the repo title seems a bit redundant. (Also, there's a trailing "-" in the repo name, in case that was missed, which makes it a pain to type out the directory name.)

The folder structure is hard to understand. What's the difference between "tensegrity examples" and "test examples"? Both have "statics" and "dynamics" subfolders. Both are tests, correct? And there seem to be tensegrity structures in both folders.

Your 'setup.m' file also doesn't seem to work as intended. MATLAB's "addpath" is only valid for one session, and needs to be run every time you re-open MATLAB, so it's not "run only once" if that's what you mean. (P.S., to fix this, please don't use 'savepath' to permanently change the saved path on a user's computer, it takes a long time to untangle unwanted behavior that way.)

One approach (you don't have to do this, there are other ways) is to have the path additions at the start of any particular script so the path gets re-added each time and each script is independent. That's what we do in tiso for example (https://github.com/apsabelhaus/tiso/blob/master/is_examples/2d/horizSpineMultiVert_2d.m). 

There are duplicates of files in various places with the same name. For example, in both Main/Tensegrity_Statics_Main and Test_Examples/Statics there are D_Bar_Statics.* files and Main_equilibrium_D_Bar.m, among others. Are they different? If not, the authors will need to eliminate code duplication as much as possible before publication (minimize the copy-paste.) 

Before any further review, I'd ask that the authors take another look at all this organization, eliminate duplicate files and duplicate or redundant or overlapping folder structures.

It might also be good to proactively look over how we reviewed STEDY (https://github.com/openjournals/joss-reviews/issues/1042) and some of the issues that were closed as part of their review (https://github.com/isrlab/stedy/issues?q=is%3Aissue+is%3Aclosed). 

In particular I'd point the authors to the thoughts on using MATLAB's 'help' command for documentation, which is one way to meet JOSS' documentation requirements for the core API. (https://github.com/isrlab/stedy/issues/8).

It's wonderful that you have a documentation PDF, but answering some of these questions in it - what does some piece of code do? What results are expected? - would be helpful.

I look forward to reviewing the code itself soon!

**Comment by apsabelhaus** [comment]
@ramaniitrgoyal92 while we're waiting for the official review to open, I added a first issue in the repo about organization and documentation, so there's something to start on! Also, the paper is missing references [???], probably a bug in the latex.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon add @vaishnavtv as reviewer

**Comment by whedon** [comment]
OK, @vaishnavtv is now a reviewer

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon add @ctdegroot as reviewer

**Comment by whedon** [comment]
OK, @ctdegroot is now a reviewer

**Comment by Kevin-Mattheus-Moerman** [comment]
@Jfriesen222 would you be interested in reviewing this work on tensegrity modelling for JOSS? The review process will focus on the software and this short paper.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon start review

**Comment by whedon** [comment]
OK, I've started the review over in https://github.com/openjournals/joss-reviews/issues/1613. Feel free to close this issue now!

**Comment by Kevin-Mattheus-Moerman** [comment]
The review has started over at #1613 however additional reviewers are still welcome. Contact me if interested.
